jacobian_follower/src/jacobian_follower/jacobian_follower.cpp

```diff
@@ -316,20 +361,21 @@ PlanResult JacobianFollower::moveInWorldFrame(planning_scene_monitor::LockedPlan
     vis_pub_->publish(msg);
   }
 
-  planning_scene->setCurrentState(start_state);
-  auto const traj =
-      jacobianPath3d(planning_scene, world_to_robot, jmg, tool_names, preferred_tool_orientations, tool_paths);
-  auto const reached_target = traj.getWayPointCount() == (tool_paths[0].size() - 1);
+  waypoint_command.context.planning_scene->setCurrentState(waypoint_command.start_state);
+  auto const traj = jacobianPath3d(waypoint_command.context.planning_scene, waypoint_command.context.world_to_robot,
+                                   jmg, waypoint_command.context.tool_names,
+                                   waypoint_command.preferred_tool_orientations, tools_waypoint_interpolated);
+  auto const reached_target = traj.getWayPointCount() == (tools_waypoint_interpolated[0].size() - 1);
```

Comment:
I don't understand this logic.

I agree this PR hasn't changed it though

Reply:
I've split it up and added a comment. It's gross I realize, so if you have an alternative I'm all ears

Reply:
There are two things going on:
1) "success" of jacobianPath3d is implicit (i.e. based on the length of the trajectory). yuck, but okay. `jacobianPath3D` could return the `reached_target`. Seems like `jacobianPath3D` should have the responsibility of deciding if it was successful, rather than here.
2) Why the `-1`? Assuming `traj.getWayPointCount()` returns the number of waypoints, then this logic disagrees with your comment

Reply:
actually I just checked the docs and getWayPointCount does return # waypoints so I agree this code seems wrong. But I've been using it a lot so I'm not sure what's going on...

Reply:
maybe it's because the interpolated waypoints includes the start waypoint, but the returned traj does not?

Reply:
I'm not too surprised to see a `-1` when comparing trajectory sizes, for reasons like that. At least the comment and the code should match